src/sentry/tasks/symbolication.py

```diff
@@ -93,24 +94,57 @@ def should_demote_symbolication(
 # This is f*** joke:
 # The `mock.patch` in `test_symbolication.py` will not work with a static import,
 # so we gotta import the function dynamically here. Great! Hooray!
-def get_native_symbolication_function(data: Any) -> Callable[[Symbolicator, Any], Any] | None:
+def get_native_symbolication_function(
+    data: Any, stacktraces: list[StacktraceInfo]
+) -> Callable[[Symbolicator, Any], Any] | None:
     from sentry.lang.native.processing import get_native_symbolication_function
 
-    return get_native_symbolication_function(data)
+    return get_native_symbolication_function(data, stacktraces)
 
 
-def get_symbolication_function(
+def get_symbolication_function_for_platform(
+    platform: SymbolicatorPlatform,
     data: Any,
-) -> tuple[SymbolicatorPlatform, Callable[[Symbolicator, Any], Any] | None]:
+    stacktraces: list[StacktraceInfo],
+) -> Callable[[Symbolicator, Any], Any]:
+    """Returns the symbolication function for the given platform
+    and event data."""
+
     from sentry.lang.java.processing import process_jvm_stacktraces
-    from sentry.lang.java.utils import should_use_symbolicator_for_proguard
 
-    if data["platform"] in ("javascript", "node"):
-        return SymbolicatorPlatform.js, process_js_stacktraces
-    elif data["platform"] == "java" and should_use_symbolicator_for_proguard(data.get("project")):
-        return SymbolicatorPlatform.jvm, process_jvm_stacktraces
+    if platform == SymbolicatorPlatform.js:
+        return process_js_stacktraces
+    elif platform == SymbolicatorPlatform.jvm:
+        return process_jvm_stacktraces
     else:
-        return SymbolicatorPlatform.native, get_native_symbolication_function(data)
+        symbolication_function = get_native_symbolication_function(data, stacktraces)
+        # get_native_symbolication_function already returned something in
+        # get_symbolication_platforms
+        assert symbolication_function is not None
+        return symbolication_function
+
+
+def get_symbolication_platforms(
+    data: Any, stacktraces: list[StacktraceInfo]
+) -> list[SymbolicatorPlatform]:
+    """Returns a list of Symbolicator platforms
+    that apply to this event."""
+
+    from sentry.lang.java.utils import is_jvm_event, should_use_symbolicator_for_proguard
+    from sentry.lang.javascript.utils import is_js_event
+
+    platforms = []
+
+    if should_use_symbolicator_for_proguard(data.get("project")) and is_jvm_event(
+        data, stacktraces
+    ):
+        platforms.append(SymbolicatorPlatform.jvm)
+    if is_js_event(data, stacktraces):
```

Comment:
```suggestion
    elif is_js_event(data, stacktraces):
```

Reply:
This is intentional. An event can pass more than one of these checks, and we want to return all platforms for which it does so. Example: if the event passes both `is_js_event` and `is_native_event`, we want the result to be `[js, native]`.